src/dll/Systems/ScriptSystem.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "Hook.hpp"
+#include "ISystem.hpp"
+#include "Paths.hpp"
+#include "PluginBase.hpp"
+
+class ScriptSystem : public ISystem
+{
+public:
+    ScriptSystem(const Paths& aPaths);
+
+    ESystemType GetType() final;
+
+    void Startup() final;
+    void Shutdown() final;
+
+    bool Add(std::shared_ptr<PluginBase> aPlugin, const char* path);
```

Comment:
Why not direct conversion to `std::filesystem::path`?

Reply:
I was getting a nullptr at the beginning of `Add` when using that - I couldn't figure out why, so switch to just plain `char*`.

Reply:
Hmm, was it because the arg passed to path was also nullptr or was it just throwing an exception directly?

Reply:
I experimented a lot with it the other day, but couldn't figure anything out that didn't result in a nullptr or empty path. I thought it was related to the implicit conversion to `path`, but I was able to do the same `const char` -> `std::filesystem::path` directly with my `InputLoader::Add` dllexport.

When inspecting the data, it looked like it was still a char*, so maybe there's some conversion that's not happening?

Reply:
I didn't try anything longer than a filename, but I don't think so - I'm using `InputLoader::Add(aHandle, "inputs.xml");` in Input Loader.